test/parallel/test-http-socket-encoding-error.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const common = require('../common');
+const http = require('http');
+
+const server = http.createServer().listen(0, connectToServer);
+
+server.on('connection', (socket) => {
```

Comment:
It's better to have a `common.mustCall` around the callback as the assertion happens in it.